### PR TITLE
Adding IPv6 Troubleshooting guide

### PR DESCRIPTION
**Pull request by DirectedSoul1**
This doc needs a few additional steps, for now, I have added to the best of my knowledge.

### CODE REVIEW

File: Troubleshooting_Ipv6.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
+# IPv6 Disconnected Install Troubleshooting
+
+In this document we will analyze common scenarios of errors/failures which can cuase the cluster installation to fail. 
+
+## Common failures Basic checks before Installing
+
+Below are some precautions we should follow before even attempting the OCP4.3 Ipv6 deployments.
```
Comment (mazzystr) [inline Troubleshooting_Ipv6.md]: OCP 4 IPv6 deployments

Don't pin this doc to a minor release unless we want to continuously update this doc

File: Troubleshooting_Ipv6.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
+# IPv6 Disconnected Install Troubleshooting
+
+In this document we will analyze common scenarios of errors/failures which can cuase the cluster installation to fail. 
+
+## Common failures Basic checks before Installing
+
+Below are some precautions we should follow before even attempting the OCP4.3 Ipv6 deployments. 
+
+* [Ipv4-BM-IPI](https://github.com/openshift-kni/baremetal-deploy/blob/master/install-steps.md#preparing-the-provision-node-for-openshift-install) has some pre-requisites which will be applied fpr ipv6 deployments as well.
```
Comment (mazzystr) [inline Troubleshooting_Ipv6.md]: I think it's a bad idea to link out to a GitHub repo not associated with the product.

File: Troubleshooting_Ipv6.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+Below are some precautions we should follow before even attempting the OCP4.3 Ipv6 deployments. 
+
+* [Ipv4-BM-IPI](https://github.com/openshift-kni/baremetal-deploy/blob/master/install-steps.md#preparing-the-provision-node-for-openshift-install) has some pre-requisites which will be applied fpr ipv6 deployments as well. 
+
+### The architecture for IPv6 Disconnected Install is 
+* Jump Host which acts as a DNSMasq, Registry. 
+* Worker/Provisioning node 
+* 3 Master nodes. 
+
+Before starting the installation, we need to set up a DNSmasq conf file which will assign IPv6 IP's to all the nodes on `eno3` interface.A sample file is provided below.
```
Comment (mazzystr) [inline Troubleshooting_Ipv6.md]: Going fwd there be dragons ... We are pushing lab architecture assumptions on to everyone.  This will immediately put us in contention with customer standards.

File: Troubleshooting_Ipv6.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+* [Create a Disconnected Registry](https://docs.openshift.com/container-platform/4.3/installing/install_config/installing-restricted-networks-preparations.html#installing-restricted-networks-preparations)
+
+**Note:** While Mirroring the registry, please make sure to add the additional registry.svc.io token to your pull-secret.txt file, else the mirroring process fails. [Add-Token](https://docs.openshift.com/container-platform/4.3/installing/install_config/installing-restricted-networks-preparations.html#Procedure) 
+
+After the registry is mirrored, confirm that you can access this in your disconnected environment
+
+```
+# curl -k -6 -u kni:kni https://registry.<Cluster-Name>.<Domain-Name>:5000/v2/_catalog
+{"repositories":["ocp4"]}
```
Comment (mazzystr) [inline Troubleshooting_Ipv6.md]: release name should be variable

File: Troubleshooting_Ipv6.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+```
+
+Now, we need to verify the images are served from local http service(either create an nginx local server or httpd container service) 
+
+```
+# curl -I -s http://<Jump-Host>.<Domain-Name>
+HTTP/1.1 200 OK
+
+# curl -I -s http://<Jump-Host>.<Domain-Name>/$RHCOS_OPENSTACK_URI | grep HTTP
```
Comment (mazzystr) [inline Troubleshooting_Ipv6.md]: What are we doing here?   There's no mention of RHCOS_OPENSTACK_URI anywhere else in the doc.  Please clarify

File: Troubleshooting_Ipv6.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+  cache_url: ''
+  deploy_kernel_url: http://172.22.0.3:6180/images/ironic-python-agent.kernel
+  deploy_ramdisk_url: http://172.22.0.3:6180/images/ironic-python-agent.initramfs
+  dhcp_range: 172.22.0.10,172.22.0.100
+  http_port: "6180"
+  ironic_endpoint: http://172.22.0.3:6385/v1/
+  ironic_inspector_endpoint: http://172.22.0.3:5050/v1/
+  provisioning_interface: eno1
+  provisioning_ip: 172.22.0.3/24
+  rhcos_image_url: ${RHCOS_PATH}${RHCOS_URI}
```
Comment (mazzystr) [inline Troubleshooting_Ipv6.md]: Same as above.  What are we doing here?   There's no mention of env var anywhere else in the doc.  Please clarify

File: Troubleshooting_Ipv6.md (hunk 2 above)
Comment (stbenjam) [reply Troubleshooting_Ipv6.md]: :+1:

If there's something specific for baremetal IPI and IPv6, we do have docs in this repo for that (/docs/user/metal). Ideally baremetal-deploy docs that are generally relevant should be added there.

File: Troubleshooting_Ipv6.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+
+echo "***Power Off master-0,master1,master-2****"
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ipv6-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ipv6-ip> power off
+ipmitool -I lanplus -U root -P <Password> -H <iDRAC-ipv6-ip> power off
+``` 
+
+## During the deployment
+
+As per the [issue](https://github.com/openshift/installer/pull/2449) we need to create a custom metal3-config.yaml file as shown below.
```
Comment (stbenjam) [inline Troubleshooting_Ipv6.md]: https://github.com/openshift/installer/pull/2449 is resolved, you can remove this entire section.

File: Troubleshooting_Ipv6.md (hunk 1 above)
Comment (DirectedSoul1) [reply Troubleshooting_Ipv6.md]: Ok, sure adding the changes

File: Troubleshooting_Ipv6.md (hunk 4 above)
Comment (DirectedSoul1) [reply Troubleshooting_Ipv6.md]: Added the changes, Thanks @mazzystr

File: Troubleshooting_Ipv6.md (hunk 5 above)
Comment (DirectedSoul1) [reply Troubleshooting_Ipv6.md]: Removing the $RHCOS_OPENSTACK_URI

Comment (DirectedSoul1) [reply Troubleshooting_Ipv6.md]: Removing $RHCOS_QEMU_URI and mapping them with proper env variables.

File: Troubleshooting_Ipv6.md (hunk 6 above)
Comment (DirectedSoul1) [reply Troubleshooting_Ipv6.md]: as suggested by @stbenjam removing this section entirely

File: Troubleshooting_Ipv6.md (hunk 7 above)
Comment (DirectedSoul1) [reply Troubleshooting_Ipv6.md]: Okie, sure, I'm removing this section entirely, Thanks @stbenjam

**Comment by abhinavdahiya** [comment]
I think this needs to be in docs/dev or docs/user/baremetal .

**Comment by DirectedSoul1** [comment]
> I think this needs to be in docs/dev or docs/user/baremetal .

Moving it to above suggested path.

File: docs/user/metal/Troubleshooting_Ipv6.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
+# IPv6 Disconnected Install Troubleshooting
```
Comment (abhinavdahiya) [inline docs/user/metal/Troubleshooting_Ipv6.md]: `docs/user/metal/Troubleshooting_Ipv6.md` -> `docs/user/metal/troubleshooting_ipv6.md`

It has content like 
```
## Create a Disconnected Registry

# BootstrapVM cannot download RHCOS image

## Bootstrap VM cannot boot up my nodes

... and more
```

which talk about general troubleshooting of baremetal ipi  and don't seem to be related to ipv6. So i think we should split these into generic and special ipv6 files.

File: docs/user/metal/troubleshooting_Ipv6.md [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+
+In this document we will analyze common scenarios of errors/failures which can cuase the cluster installation to fail. 
+
+## Common failures Basic checks before Installing
+
+Below are some precautions we should follow before even attempting the OCP4 Ipv6 deployments. 
+
+* [Ipv4-BM-IPI](https://github.com/openshift-kni/baremetal-deploy/blob/master/install-steps.md#preparing-the-provision-node-for-openshift-install) has some pre-requisites which will be applied fpr ipv6 deployments as well. 
+
+### The architecture for IPv6 Disconnected Install(Nodes can vary as per requirements)
```
Comment (stbenjam) [inline docs/user/metal/troubleshooting_Ipv6.md:11]: These seem specific to a development/testing environment. I don't think I'd want these documented here

File: docs/user/metal/troubleshooting_Ipv6.md [hunk 10]
```diff
@@ -0,0 +1,328 @@
+# IPv6 Disconnected Install Troubleshooting
+
+In this document we will analyze common scenarios of errors/failures which can cuase the cluster installation to fail. 
+
+## Common failures Basic checks before Installing
+
+Below are some precautions we should follow before even attempting the OCP4 Ipv6 deployments. 
+
+* [Ipv4-BM-IPI](https://github.com/openshift-kni/baremetal-deploy/blob/master/install-steps.md#preparing-the-provision-node-for-openshift-install) has some pre-requisites which will be applied fpr ipv6 deployments as well.
```
Comment (stbenjam) [inline docs/user/metal/troubleshooting_Ipv6.md:9]: Link is stale

**Comment by rlopez133** [comment]
> Looks like feedback is still unaddressed - file needs renaming for one (docs/user/metal/Troubleshooting_Ipv6.md -> docs/user/metal/troubleshooting_ipv6.md). A lot of this is specific to development environments or is general troubleshooting not specific to IPv6. I also think a lot of this could go to baremetal-deploy repo's docs.
> 
> Do you plan to revisit this? It's about to get closed.

This was opened before we had all the docs we have in the baremetal-deploy repo. Since this was PR'ed, a lot of changes have been put in place. In fact, we are in the process of updating the troubleshooting docs even further in the baremetal-deploy repo.
